source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -56,11 +56,19 @@ Http::FilterHeadersStatus
 RoleBasedAccessControlFilter::decodeHeaders(Http::RequestHeaderMap& headers, bool) {
   ENVOY_LOG(
       debug,
-      "checking request: requestedServerName: {}, remoteAddress: {}, localAddress: {}, ssl: {}, "
-      "headers: {}, dynamicMetadata: {}",
+      "checking request: requestedServerName: {}, sourceIP: {}, peerIP: {}, remoteIP: {},"
+      "localAddress: {}, ssl: {}, headers: {}, dynamicMetadata: {}",
       callbacks_->connection()->requestedServerName(),
       callbacks_->connection()->remoteAddress()->asString(),
-      callbacks_->connection()->localAddress()->asString(),
+      callbacks_->streamInfo().downstreamDirectRemoteAddress()
+          ? callbacks_->streamInfo().downstreamDirectRemoteAddress()->asString()
+          : "",
+      callbacks_->streamInfo().downstreamRemoteAddress()
```

Comment:
Instead of checking for null in all of these calls, consider making the tests set a valid address. Same in log message in network filter.

Reply:
It should not be null according to its comment. I changed to check the null because it only happens in the coverage test of RBACIntegrationTest in rbac_filter_integration_test.cc, other tests doesn't have the issue (the same test pass in non-coverage test). Do you have any idea how to fix this? Thanks.

Reply:
There was a recent change (this morning) that fixed some issues related to filter registration in the coverage build; I'm not sure if it's related or not. I tried pulling your branch and merging master, and coverage passed for me. It *looks* like all the codepaths should be fine without the null checks. Can you try it once more against master without the null checks? If it fails again, leave them in because they should be harmless.